pydax/_schema_retrieval.py

```diff
@@ -36,18 +37,26 @@ def retrieve_schema_file(url_or_path: Union[_typing.PathLike, str], encoding: st
     :raises ValueError: An error occurred when parsing `url_or_path` as either a URL or path.
     :return: A string of the content.
     """
+
     url_or_path = str(url_or_path)
-    scheme = urlparse(url_or_path).scheme
-
-    if scheme in ('http', 'https'):
-        content = requests.get(url_or_path, allow_redirects=True).content
-        # We don't use requests.Response.encoding and requests.Response.text because it is always silent when there's an
-        # encoding error
-        return content.decode(encoding)
-    elif scheme == 'file':
-        with urlopen(url_or_path) as f:  # nosec: bandit will always complain but we know the URL points to a local file
-            return f.read().decode(encoding)
-    elif scheme == '':  # local file path
-        return Path(url_or_path).read_text(encoding)
+
+    # We don't detect fully whether the input is a URL or a file path because I couldn't find a reliable way. Almost any
+    # string with no backslash can be a file name on Linux. URL detection often involves either giant dependencies such
+    # as Django, or tediously long regular expression that we can't assure that it would work. Here, we detect the
+    # beginning of the string. If it doesn't look like a URL, treat it as a file path.
```

Comment:
What about the validators dependency? https://validators.readthedocs.io/en/latest/#module-validators.url Looking at the source code it seems like it is using a tedious regular expression.

Reply:
I've looked into it, but I don't think it's worth the effort for (1) such a single use to depend on (2) a library with this level of support (it's not actively maintained, and not backed by a community, despite a decent amount of users).

Reply:
ok got it, thanks for laying out the reasons